### PR TITLE
Accept IBoxService in BoxJWTAuth constructor

### LINKED ISSUE #544: BoxJWTAuth does not accept custom IBoxService in constructor
- [x] I have checked that the [SDK documentation][sdk-docs] and [API documentation][api-docs] doesn't solve my issue

### Description of the Issue
BoxJWTAuth does not accept IBoxService in constructor. Providing a custom IBoxService could help to log tracing information and to add custom retry logic by implementing custom IRequestHandler.

### Versions Used
.NET SDK: v3.13.1
.NET Framework/Core: .Net Core 2.2.3

[sdk-docs]: ./README.md
[api-docs]: https://developer.box.com/docs

### PR DESCRIPTION
**Pull request by kostya9**
Fixes #544

Motivation: I would like to be able to add custom IRequestHandlers to GetToken requests, so that I would be able to, e.g. log some relevant information

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @kostya9 has signed the CLA. Thanks for the pull request!

File: Box.V2/JWTAuth/BoxJWTAuth.cs [hunk 1]
```diff
@@ -83,6 +87,15 @@ public BoxJWTAuth(IBoxConfig boxConfig)
             }
         }
 
+        /// <summary>
+        /// Constructor for JWT authentication with default boxService
+        /// </summary>
+        /// <param name="boxConfig">Config contains information about client id, client secret, enterprise id, private key, private key password, public key id </param>
+        public BoxJWTAuth(IBoxConfig boxConfig) : this(boxConfig, new BoxService(new HttpRequestHandler()))
```
Comment (kostya9) [inline Box.V2/JWTAuth/BoxJWTAuth.cs]: This constructor was added in order to preserve backward compatibility

**Comment by mattwiller** [comment]
@kostya9 Thanks for the pull request!  Your code looks good to me, but that tests are failing so I won't be able to merge your changes just yet.  I can't tell if the test failures are related to your code — I'll investigate and merge your code if it turns out they're unrelated.

**Comment by kostya9** [comment]
@mattwiller Thanks for checking the tests! What is the process of releasing new version of Box.V2.Core Nuget package? I wonder when can I expect the changes to be published

**Comment by mattwiller** [comment]
@kostya9 We just released v3.15.0 of the SDK, which includes your change.  Please give it a try and let us know if you run into any further issues.  Thanks again for the PR! 🎉

**Pull request merged**